polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -189,10 +205,19 @@ class ScalaCompiler private (
     compilationUnit: RichCompilationUnit = new global.RichCompilationUnit(NoSourceFile),
     sourceFile: SourceFile = NoSourceFile
   ) {
+
     // this copy of the code will be mutated by compile
-    lazy val compiledCode: List[Tree] = code.map {
-      stat => stat.duplicate.setPos(stat.pos)
-    }
+    lazy val compiledCode: List[Tree] = code.foldLeft(List.empty[Tree]) {
```

Comment:
Here we're just making sure any unpositioned subtrees (e.g. those injected by the interpreter) have a transparent position that won't interfere with anything (but still make as much sense as possible in case those trees cause an error)